lib/models/file-patch.js

```diff
@@ -208,7 +225,16 @@ export default class FilePatch {
 
   getUnstagePatchForLines(selectedLines) {
     if (this.changedLineCount === [...selectedLines].filter(line => line.isChanged()).length) {
-      return this.getUnstagePatch();
+      if (this.hasTypechange() && this.getStatus() === 'added') {
+        // handle special case when a file was created after a symlink was deleted.
+        // In order to unstage the file creation, we must ensure that the unstage patch has no new file,
+        // so when the patch is applied to the index, there file will be removed from the index
+        return this.getUnstagePatch().clone({
+          newFile: new File({path: null, mode: null, symlink: null}),
+        });
+      } else {
+        return this.getUnstagePatch();
+      }
```

Comment:
This look great! If they're not exercised in a repo test, we should consider adding specific `FilePatch` tests for them.

Reply:
Done!